doc/manual_experimental.rst

```diff
@@ -418,1296 +429,1139 @@ here.
 
-.. code-block:: nim
+    type
+      IntEnumerable = vtref Enumerable[int]
 
-   `&&`("my", space & "awe", "some ", "concat")
+      MyObject = object
+        enumerables: seq[IntEnumerable]
+        streams: seq[OutputStream.vtref]
 
+    proc addEnumerable(o: var MyObject, e: IntEnumerable) =
+      o.enumerables.add e
 
-The `**` operator
-~~~~~~~~~~~~~~~~~~~
+    proc addStream(o: var MyObject, e: OutputStream.vtref) =
+      o.streams.add e
 
-The `**` is much like the `*` operator, except that it gathers not only
-all the arguments, but also the matched operators in reverse polish notation:
+  The procs that will be included in the vtable are derived from the concept
+  body and include all proc calls for which all param types were specified as
+  concrete types. All such calls should include exactly one param of the type
+  matched against the concept (not necessarily in the first position), which
+  will be considered the value bound to the vtable.
 
-.. code-block:: nim
+  Overloads will be created for all captured procs, accepting the vtable type
+  in the position of the captured underlying object.
 
-  import std/macros
+  Under these rules, it's possible to obtain a vtable type for a concept with
+  unbound type parameters or one instantiated with metatypes (type classes),
+  but it will include a smaller number of captured procs. A completely empty
+  vtable will be reported as an error.
 
-  type
-    Matrix = object
-      dummy: int
+  The `vtref` magic produces types which can be bound to `ref` types and
+  the `vtptr` magic produced types bound to `ptr` types.
 
-  proc `*`(a, b: Matrix): Matrix = discard
-  proc `+`(a, b: Matrix): Matrix = discard
-  proc `-`(a, b: Matrix): Matrix = discard
-  proc `$`(a: Matrix): string = result = $a.dummy
-  proc mat21(): Matrix =
-    result.dummy = 21
 
-  macro optM{ (`+`|`-`|`*`) ** a }(a: Matrix): untyped =
-    echo treeRepr(a)
-    result = newCall(bindSym"mat21")
+Type bound operations
+=====================
 
-  var x, y, z: Matrix
+There are 4 operations that are bound to a type:
 
-  echo x + y * z - x
+1. Assignment
+2. Moves
+3. Destruction
+4. Deep copying for communication between threads
 
-This passes the expression `x + y * z - x` to the `optM` macro as
-an `nnkArgList` node containing::
+These operations can be *overridden* instead of *overloaded*. This means the
+implementation is automatically lifted to structured types. For instance if type
+`T` has an overridden assignment operator `=` this operator is also used
+for assignments of the type `seq[T]`. Since these operations are bound to a
+type they have to be bound to a nominal type for reasons of simplicity of
+implementation: This means an overridden `deepCopy` for `ref T` is really
+bound to `T` and not to `ref T`. This also means that one cannot override
+`deepCopy` for both `ptr T` and `ref T` at the same time; instead a
+helper distinct or object type has to be used for one pointer type.
```

Comment:
```suggestion
These operations can be *overridden* instead of *overloaded*. This means the
implementation is automatically lifted to structured types. For instance, if type
`T` has an overridden assignment operator `=`, this operator is also used
for assignments of the type `seq[T]`. Since these operations are bound to a
type, they have to be bound to a nominal type for reasons of simplicity of
implementation: This means an overridden `deepCopy` for `ref T` is really
bound to `T` and not to `ref T`. This also means that one cannot override
`deepCopy` for both `ptr T` and `ref T` at the same time; instead a
distinct or object helper type has to be used for one pointer type.
```